engine/src/main/java/com/sandboni/core/engine/filter/VertexFilter.java

```diff
@@ -0,0 +1,14 @@
+package com.sandboni.core.engine.filter;
+
+import com.sandboni.core.engine.sta.graph.Link;
+import com.sandboni.core.engine.sta.graph.LinkType;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Stream;
+
+public interface VertexFilter {
+
+    Set<Vertex> filter(ConcurrentHashMap<LinkType, Boolean> adoptedLinkTypes, Stream<Link> allLinks);
```

Comment:
Do you really need to define a specific implementation here? maybe a Map<> could be more appropriate value for an interface?

Reply:
fixed